src/operator/gitea.ts

```diff
@@ -480,11 +480,18 @@ async function setGiteaOIDCConfig() {
     const execCommand = [
       'sh',
       '-c',
-      `AUTH_ID=$(gitea admin auth list --vertical-bars | grep -E "\\|otomi-idp\\s+\\|" | grep -iE "\\|OAuth2\\s+\\|" | awk -F " " '{print $1}' | tr -d '\\n') && gitea admin auth update-oauth --id "$AUTH_ID" --key "${clientID}" --secret "${clientSecret}" --auto-discover-url "${discoveryURL}"`,
+      `
+      AUTH_ID=$(gitea admin auth list --vertical-bars | grep -E "\\|otomi-idp\\s+\\|" | grep -iE "\\|OAuth2\\s+\\|" | awk -F " " '{print $1}' | tr -d '\\n')
```

Comment:
This operator should periodically check if oauth setting are in place.

Consider the following use cases:
1. The gitea and gitea-operator are in the running state. However the DNS is not ready yet. What will happen then?
2. Someone remove otomi-idp from Gitea. The operator will recreated but only if it gets restarted

Reply:
Thanks for implementing retry mechanism